src/api.c

```diff
@@ -2714,6 +2714,46 @@ static int cgroup_get_parent_name(struct cgroup *cgroup, char **parent)
 	return ret;
 }
 
+/*
+ * Checks if the cgroup's controller shares the mount point with any other
+ * controller of the cgroup. Returns 1 if shared or 0.
+ */
+static int is_cgrp_ctrl_mnt_shared(struct cgroup *cgroup, char *controller)
```

Comment:
I think if we go with this function, we really need to unit test it.  Should be fairly easy to do, and it should allow us to catch corner cases.